### PR TITLE
fixed union bank scraper

### PR DESCRIPTION
**Pull request by orentet**
also, please add me to the contributors of union bank scraper.

### CODE REVIEW

File: src/scrapers/union-bank.ts [hunk 1]
```diff
@@ -210,7 +210,7 @@ async function searchByDates(page: Page, startDate: Moment) {
 
 async function getAccountNumber(page: Page) {
   const selectedSnifAccount = await page.$eval('#ddlAccounts_m_ddl option[selected="selected"]', (option) => {
-    return (option as HTMLElement).innerText;
+    return (option as HTMLElement).innerText.replace(/[^ -~]+/g, '');
```
Comment (eshaham) [inline src/scrapers/union-bank.ts:213]: @orentet why is this needed?
missing an explanation for this change in your PR

File: src/scrapers/union-bank.ts [hunk 2]
```diff
@@ -274,10 +274,7 @@ async function fetchAccounts(page: Page, startDate: Moment) {
 }
 
 async function waitForPostLogin(page: Page) {
-  return Promise.race([
-    waitUntilElementFound(page, '#signoff', true),
-    waitUntilElementFound(page, '#restore', true),
-  ]);
+  return waitUntilElementFound(page, '.signoff_button', false);
```
Comment (eshaham) [inline src/scrapers/union-bank.ts:277]: @orentet why are we changing this logic? doesn't it work as it is?

**Review by eshaham — CHANGES_REQUESTED** [review]
@orentet thanks for your contribution
can you please add a description to your PR to explain your changes?

File: src/scrapers/union-bank.ts (hunk 2 above)
Comment (orentet) [reply src/scrapers/union-bank.ts:277]: no. the UI was probably changed.

File: src/scrapers/union-bank.ts (hunk 1 above)
Comment (orentet) [reply src/scrapers/union-bank.ts:213]: when selecting the account, it contains non-printable chars on both sides, which this code deletes.

**Comment by orentet** [comment]
any change?

**Comment by esakal** [comment]
@orentet Hi,
the PR is missing a description and it is not clear from the code change why you introduce those changes. Please add description explaining your changes. Thanks

**Comment by orentet** [comment]
Hello Eran,
Please look at my responses for the code changes.
The reason for the changes is because union bank is currently not working
because of UI changes, and my code fixes it.

File: src/scrapers/union-bank.ts (hunk 1 above)
Comment (esakal) [reply src/scrapers/union-bank.ts:213]: maybe I'm wrong but when using `[^ -~]+` pattern (starting with ^ symbol in the brackets) you are basically replacing everything that is not one of those characters, so you keep those signs and not the actual account value. where am I mistaken?

regarding that non-printable sign, I tried your regexp in two online editors and that sign broke them. can you instead use the hex representation of that sign, or maybe use allowed list instead. for example `/[\d-]+/g`

File: src/scrapers/union-bank.ts [hunk 3]
```diff
@@ -274,10 +274,7 @@ async function fetchAccounts(page: Page, startDate: Moment) {
 }
 
 async function waitForPostLogin(page: Page) {
-  return Promise.race([
-    waitUntilElementFound(page, '#signoff', true),
-    waitUntilElementFound(page, '#restore', true),
```
Comment (esakal) [inline src/scrapers/union-bank.ts:279]: why did you remove the #restore part? this is basically the view that is shown once in a while to replace a password? I don't have union so I cannot verify it but based on other scrapers usually here the scraper wait for either a valid login or a password expired status hence the promise.race

consider keeping the restore part

**Comment by esakal** [comment]
@orentet thanks for the clarification. The Github code review dashboard is extremely annoying and misleading. In my workplace we use [https://reviewable.io/](https://reviewable.io/) which is must superior from the Github one.  Please don't resolve the comments, just add your insights on them, the resolution should be done only once we all agree.

I added my insights about your changes, let me know what you thinkg
